libs/agno/agno/db/sqlite/sqlite.py

```diff
@@ -505,43 +503,18 @@ def rename_session(
             Exception: If an error occurs during renaming.
         """
         try:
-            table = self._get_table(table_type="sessions")
-            if table is None:
+            # Get the current session as a deserialized object
+            session = self.get_session(session_id, session_type, deserialize=True)
+            if session is None or not isinstance(session, (AgentSession, TeamSession, WorkflowSession)):
                 return None
 
```

Comment:
```suggestion
            # Get the session record
            session = self.get_session(session_id, session_type, deserialize=True)
            if session is None:
                return None

            session = cast(Session, session)
```

Checking the Session types could break later if we add new ones. This is a bit more resilient. You'll need to `from typing import cast` at the top :)